gateway/src/test/java/org/georchestra/gateway/security/accessrules/AccessRulesCustomizerIT.java

```diff
@@ -0,0 +1,330 @@
+                .expectStatus().isFound()//
+                .expectHeader().location("/login");
+    }
+
+    /**
+     * <pre>
+     * {@code
+     * georchestra.gateway.services.import: 
+     *   access-rules:
+     *   - intercept-url: /import/**
+     *     anonymous: false
+     * }
+     */
+    @WithMockUser(authorities = { "ROLE_DOESNTMATTER" })
+    public @Test void testService_requires_any_authenticated_user() {
+        mockService.stubFor(get(urlMatching("/import(/.*)?")).willReturn(ok()));
+
+        testClient.get().uri("/import")//
+                .exchange()//
+                .expectStatus().isOk();
+
+        testClient.get().uri("/import/any/thing")//
+                .exchange()//
+                .expectStatus().isOk();
+    }
+
+    /**
+     * <pre>
+     * {@code
+     * georchestra.gateway.services.analytics: 
+     *   access-rules:
+     *   - intercept-url: /analytics/**
+     *     allowed-roles: SUPERUSER,ORGADMIN
+     * }
+     */
+    @WithMockUser(authorities = { "ROLE_USER", "ROLE_EDITOR" })
+    public @Test void testService_requires_specific_role_forbidden_for_non_matching_roles() {
+        mockService.stubFor(get(urlMatching("/analytics(/.*)?")).willReturn(ok()));
+
+        testClient.get().uri("/analytics")//
+                .exchange()//
+                .expectStatus().isForbidden();
+
+        testClient.get().uri("/analytics/any/thing")//
+                .exchange()//
+                .expectStatus().isForbidden();
+    }
+
+    /**
+     * <pre>
+     * {@code
+     * georchestra.gateway.services.analytics: 
+     *   access-rules:
+     *   - intercept-url: /analytics/**
+     *     allowed-roles: SUPERUSER,ORGADMIN
+     * }
+     */
+    @WithMockUser(authorities = { "ROLE_USER", "ROLE_ORGADMIN" })
+    public @Test void testService_requires_specific_role_allowed_for_matching_roles() {
+        mockService.stubFor(get(urlMatching("/analytics(/.*)?")).willReturn(ok()));
+
+        testClient.get().uri("/analytics")//
+                .exchange()//
+                .expectStatus().isOk();
+
+        testClient.get().uri("/analytics/any/thing")//
+                .exchange()//
+                .expectStatus().isOk();
+    }
+
+    /**
+     * <pre>
+     * {@code
+     * georchestra.gateway.services.analytics: 
+     *   access-rules:
+     *   - intercept-url: /analytics/**
+     *     allowed-roles: SUPERUSER,ORGADMIN
+     * }
+     */
+    public @Test void testService_requires_specific_roles_redirects_unauthenticated_to_login() {
+        mockService.stubFor(get(urlMatching("/analytics(/.*)?")).willReturn(ok()));
+
+        testClient.get().uri("/analytics")//
+                .exchange()//
+                .expectStatus().isFound()//
+                .expectHeader().location("/login");
+
+        testClient.get().uri("/analytics/any/thing")//
+                .exchange()//
+                .expectStatus().isFound()//
+                .expectHeader().location("/login");
```

Comment:
which mimics what would happen in real life. If you hit /analytics/anything in the actual analytics app, it'd return OK. But we want to assert the request doesn't even get there because the access rule requires a given role, and there's no authenticated user, so it gets redirected to the login page?